src/devtools/client/debugger/src/components/Editor/DebugLine.js

```diff
@@ -114,6 +115,12 @@ const mapStateToProps = state => {
   const frame = getVisibleSelectedFrame(state);
   const previewLocation = getPausePreviewLocation(state);
   const location = previewLocation || (frame && frame.location);
+  if (location) {
+    location.sourceId = ThreadFront.pickCorrespondingSourceId(
+      location.sourceId,
+      location.sourceUrl
+    );
+  }
```

Comment:
We should not be using `ThreadFront` in our React components. What is this trying to do?

Reply:
Good point, I moved the logic from the connector to a selector.

> What is this trying to do?

If there are multiple identical sources, it switches to the one that is [shown by the editor](https://github.com/RecordReplay/devtools/blob/e61fc29444acebff066ea3b24419d8f36ea4d2ec/src/devtools/client/debugger/src/actions/sources/select.js#L107-L109).